### ISSUE TITLE
No source .zip file available for 1.13.2 on PyPi

### ISSUE THREAD

**Issue by sashkab** [numpy/numpy#9787]
A sof 9:09amEDT on 9/28/2017, there is no numpy-1.13.2.zip available on PyPi. Only wheels for various platforms.

<img width="1626" alt="screenshot_2017 09 28_091007" src="https://user-images.githubusercontent.com/868842/30968302-df7d3cac-a42c-11e7-9d62-c37f319aa2ec.png">

Please consider uploading source as well.

**Comment by sashkab** [comment]
Here is failure attempting to build wheel myself:

```sh
$ pip wheel --no-binary numpy numpy==1.13.2
Collecting numpy==1.13.2
  Could not find a version that satisfies the requirement numpy==1.13.2 (from versions: 1.3.0, 1.4.1, 1.5.0, 1.5.1, 1.6.0, 1.6.1, 1.6.2, 1.7.0, 1.7.1, 1.7.2, 1.8.0, 1.8.1, 1.8.2, 1.9.0, 1.9.1, 1.9.2, 1.9.3, 1.10.0.post2, 1.10.1, 1.10.2, 1.10.4, 1.11.0b3, 1.11.0rc1, 1.11.0rc2, 1.11.0, 1.11.1rc1, 1.11.1, 1.11.2rc1, 1.11.2, 1.11.3, 1.12.0b1, 1.12.0rc1, 1.12.0rc2, 1.12.0, 1.12.1rc1, 1.12.1, 1.13.0rc1, 1.13.0rc2, 1.13.0, 1.13.1)
No matching distribution found for numpy==1.13.2
```

**Comment by charris** [comment]
There is a bug in cython 2.7 which affects Python 3.5, so I removed the source file until we can upload a new version. The source files contained code cythonized with 2.7.

**Comment by charris** [comment]
I've uploaded a updated file built with cython 0.26.1, could you check that it works?

**Comment by sashkab** [comment]
It doesn't when asking for 1.13.2, but when asked for 1.13.2-1 it did with a warning:

```
$ pip wheel --no-binary numpy numpy==1.13.2
Collecting numpy==1.13.2
  Could not find a version that satisfies the requirement numpy==1.13.2 (from versions: 1.3.0, 1.4.1, 1.5.0, 1.5.1, 1.6.0, 1.6.1, 1.6.2, 1.7.0, 1.7.1, 1.7.2, 1.8.0, 1.8.1, 1.8.2, 1.9.0, 1.9.1, 1.9.2, 1.9.3, 1.10.0.post2, 1.10.1, 1.10.2, 1.10.4, 1.11.0b3, 1.11.0rc1, 1.11.0rc2, 1.11.0, 1.11.1rc1, 1.11.1, 1.11.2rc1, 1.11.2, 1.11.3, 1.12.0b1, 1.12.0rc1, 1.12.0rc2, 1.12.0, 1.12.1rc1, 1.12.1, 1.13.0rc1, 1.13.0rc2, 1.13.0, 1.13.1, 1.13.2.post1)
No matching distribution found for numpy==1.13.2

$ pip wheel --no-binary numpy numpy==1.13.2-1
Collecting numpy==1.13.2-1
  Downloading numpy-1.13.2-1.zip (5.0MB)
    100% |████████████████████████████████| 5.0MB 85kB/s
  Requested numpy==1.13.2-1 from https://pypi.python.org/packages/12/41/5257cc2386948f7e1f593ded0ea1d16f5e75fdc54994334fe05bf0c4d315/numpy-1.13.2-1.zip#md5=94c48ee8dc28793032a591c096130b84, but installing version 1.13.1
```

**Comment by charris** [comment]
What happens if you don't specify the numpy version?

**Comment by charris** [comment]
Or leave out the wheel?

**Comment by papachoco** [comment]
We may need a new small bump release.  In legacy installations we get the following error
File ".../Documents/workspace/nti.dataserver-buildout/eggs/zc.buildout-2.9.5-py2.7.egg/zc/buildout/easy_install.py", line 563, in _get_dist
    dists = [_move_to_eggs_dir_and_compile(dist, self._dest)]
  File ".../Documents/workspace/nti.dataserver-buildout/eggs/zc.buildout-2.9.5-py2.7.egg/zc/buildout/easy_install.py", line 1730, in _move_to_eggs_dir_and_compile
assert newdist is not None  # newloc above is missing our dist?!
AssertionError

Also when using pip .. it seems to have running setup.py

**Comment by charris** [comment]
`pip download numpy` seems to work to get the source release.

**Comment by papachoco** [comment]
FWIW .. pip install --upgrade numpy was the command I was using when it hanged

**Comment by charris** [comment]
@papachoco This seems to work with pip 9.0.1 when no numpy is installed.
```
charris@fc [~]$ pip install --no-binary :all: --upgrade numpy --user 
Collecting numpy
  Using cached numpy-1.13.2-1.zip
Skipping bdist_wheel for numpy, due to binaries being disabled for it.
Installing collected packages: numpy
  Running setup.py install for numpy ... done
Successfully installed numpy-1.13.2
```
```

**Comment by charris** [comment]
@sashkab What pip version do you have?

**Comment by sashkab** [comment]
@charris 9.0.1. It works correctly without version specification, i.e. `pip wheel --no-binary :all:  numpy`, and when I loosely specified to use 1.13.* `pip wheel --no-binary :all:  "numpy>=1.13.0,<1.14.0"`.

**Comment by matthew-brett** [comment]
This seems to me to be a bug in pip - that it doesn't read through the build tag for the source file.   I'll ask over on the distutils-sig list.

**Comment by matthew-brett** [comment]
Ah no - I see - build tags are specific to wheel filenames : https://www.python.org/dev/peps/pep-0425/#use - so the source archive behavior is expected, I think.

**Comment by sashkab** [comment]
I think this can be closed now, as 1.13.3 was released and still available.
